packages/zoe/src/contractFacet/zcfSeat.js

```
@@ -289,6 +300,91 @@ export const createSeatManager = (
+            allocations.flatMap(Object.values);
+          const previousAmounts = flattenAllocations(
+            newAllocations.map(([seat]) => seat.getCurrentAllocation()),
+          );
+          const newAmounts = flattenAllocations(
+            newAllocations.map(([_, allocation]) => allocation),
+          );
+          assertRightsConserved(previousAmounts, newAmounts);
+
+          // ////// Ensure that offer safety holds ///////////////////////
+          newAllocations.forEach(([seat, allocation]) => {
+            isOfferSafe(seat.getProposal(), allocation) ||
+              Fail`Offer safety was violated by the proposed allocation: ${allocation}. Proposal was ${seat.getProposal()}`;
+          });
+
+          const seatHandleAllocations = newAllocations.map(
+            ([seat, allocation]) => {
+              const seatHandle = zcfSeatToSeatHandle.get(seat);
+              return { seatHandle, allocation };
+            },
+          );
+          try {
+            // No side effects above. All conditions checked which could have
+            // caused us to reject this reallocation. Notice that the current
+            // allocations are captured in seatHandleAllocations, so there must
+            // be no awaits between that assignment and here.
+            //
+            // COMMIT POINT
+            //
+            // The effects must succeed atomically. The call to
+            // replaceAllocations() will be processed in the order of updates
+            // from zcf to zoe, its effects must occur immediately in zoe on
+            // reception, and must not fail.
+            //
+            // Commit the new allocations (currentAllocation is replaced
+            // for each of the seats) and inform Zoe of the new allocation.
+
+            newAllocations.forEach(([seat, allocation]) =>
+              activeZCFSeats.set(seat, allocation),
+            );
+
+            E(zoeInstanceAdmin).replaceAllocations(seatHandleAllocations);
```

this is a promise that may reject. if so it will fail silently. 

I see that `reallocate` had this same problem. Am I missing something?

---

We have tried to ensure that ZCF has the authoritative word here. If ZCF and Zoe lose contact then things are very broken and will shortly fall apart.

Short of that, Zoe and ZCF are in close communication and we have tried to ensure that the two sides remain in sync, and for every shared bit of info, we know which side is authoritative.

---

Ok. Please put `void` and a comment with that justification. In particular, why a failure of `replaceAllocations` shouldn't be caught by the catch and `shutdownWithFailure`.

---

done